Feature ramsete auto

AntarcticaByToto opened the pull request:
Intake is currently commented out because it would need the auto/teleop state changes

AntarcticaByToto committed:
Implemented current Auto code from last year. Error with intake solenoid.

AntarcticaByToto committed:
working on changes. makign valor subsytem changes. Then going to rebase again

AntarcticaByToto committed:
all auto pathing and commands for drivetrain implemented. Will want to run drive charcterization. Intake disabled because it needs the Valor subsystem Auto/teleop state switch to be implemented

janhelgeson committed:
Merge branch 'dev' into feature-ramsete-auto

janhelgeson commented:
@mray190 Please review the merge conflict resolutions to make sure they look ok.

mray190 committed:
Synced old subsystems with new changes to ValorSubsystem

mray190 committed:
Re-factored auto

Removed un-necessary classes
Cleaned up last years code for ease of use
Provided a framework for new autos to be created

mray190 committed:
Further refactor and simplification

mray190 committed:
Shooter turret turning
